### PR TITLE
Building an image classification model with Gradio and Keras

### LINKED ISSUE #7241: [Machine Learning] Building an image classification model with Gradio and Keras
#### Proposed title of article
[Machine Learning] Building an image classification model with Gradio and Keras

#### Proposed article introduction
Image classification is one of the supervised machine learning problems which aims to categorize the images of a dataset into their respective categories or labels.  Image classification is applied in computer vision to categorize an image into three or more classes.

Gradio is an open-source python library that allows you to quickly create easy-to-use, customizable UI components for your machine learning model. Gradio allows you to integrate the GUI directly into your Python notebook making it easier to use. Gradio can also be embedded in Jupyter notebooks and Google Colab notebooks. This does not require running a separate Python script. A user can interact with the model right in the working environment.

It also enables you to create demos of your machine learning model. These demos can be used to present ideas to clients, users, and team members before the actual application are implemented.

Keras is an open-source software library that provides a Python interface for artificial neural networks. Keras acts as an interface for the TensorFlow library. We will use Keras to build the image classification model and Gradio to create the model interface.

#### Key takeaways 
1. Benefits of using Gradio
2. Building a Simple Gradio interface.
3. Image preprocessing techniques in Python.
4. Initializing the Keras layers.
5. Fitting and training the image classification model.
6. Launching the Gradio interface in Google Colab.
7. Using the model to classify images of flowers and animals.
#### Article quality
This tutorial is unique because we will implement a custom image classification model from scratch.  We build an a model that classifies images of flowers and animals.  We will explain all the step of build and adding the Keras layers so that a beginner can easily follow. We will also cover all the steps involved in image preprocessing. It is a crucial step in building an image classification model. We will use `tf.keras.preprocessing` function for image classification.

After implementing the model, we will use Gradio to build a beautiful user interface. The interface will enable users to interact with the model and visualize the classification resulst. 
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by elishanjeche**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7241

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by lalith1403** [comment]
@Dawe-7 Please focus on upto 2 PRs at a time. You have assigned 6 PRs currently.

**Comment by lalith1403** [comment]
As per our conversation, please abide by the community rules. Assigning 8 PRs at a time results in a quality decrease.

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers is anyone available to finish up the tech review on this PR.

**Comment by Dawe-7** [comment]
> @section-engineering-education/peer-reviewers is anyone available to finish up the tech review on this PR.

As per our conversation with @hectorkambow via slack DM, i will be finishing up this PR's review.
CC : @lalith1403, @louisefindlay23.

**Review by Dawe-7 — APPROVED** [review]
Awesome article @elishanjeche. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles.

**Review by Kayere — APPROVED** [review]
Nice article @elishanjeche.
Thank you for the review @Dawe-7.
